### DISCUSSION TITLE
Laravel Passport: PKCE Flow: Authorization page does not always show up / is inconsistent

### DISCUSSION THREAD

**Discussion by prosticoco** [laravel/framework#37433] [Q&A]
Hello, I am trying to implement the PKCE flow and so far I have been able to get tokens. However, the way the redirection works are inconsistent: When I start the PKCE flow (from a react-native android app), the laravel server redirects me to the login page (correct so fare), however after I log in, then sometimes (I do not know what parameter influences this event) I am redirected to the default home page (and therefore I cannot complete the PKCE flow) and some other times, I am correctly redirected to the authorization page (where consent to access the protected API resource is given by the user). I don't understand why this behavior is inconsistent and how to make such that the user is always redirected to the consent page. 

consent page : 
![image](https://user-images.githubusercontent.com/25950732/118998733-f8289780-b989-11eb-8564-e65f91aaa366.png)

It seems this issue is related to the fact that authenticated users are usually redirected to the home page when they log in outside of a PKCE flow

Thanks in advance for the help and let me know if you need more info

**Comment by hafezdivandari** [comment]
Check the added `prompt` parameter on Passport 11.1 - 11.3 when redirecting for authorization: https://laravel.com/docs/9.x/passport#requesting-tokens-redirecting-for-authorization

**Comment by InputOutputZ** [comment]
> Hello, I am trying to implement the PKCE flow and so far I have been able to get tokens. However, the way the redirection works are inconsistent: When I start the PKCE flow (from a react-native android app), the laravel server redirects me to the login page (correct so fare), however after I log in, then sometimes (I do not know what parameter influences this event) I am redirected to the default home page (and therefore I cannot complete the PKCE flow) and some other times, I am correctly redirected to the authorization page (where consent to access the protected API resource is given by the user). I don't understand why this behavior is inconsistent and how to make such that the user is always redirected to the consent page.
> 
> consent page : ![image](https://user-images.githubusercontent.com/25950732/118998733-f8289780-b989-11eb-8564-e65f91aaa366.png)
> 
> It seems this issue is related to the fact that authenticated users are usually redirected to the home page when they log in outside of a PKCE flow
> 
> Thanks in advance for the help and let me know if you need more info

I think I found the answer, for the inconsistent bit after spending many hours analysing what is going on. It lies in the AuthorizationController, approveRequest if condition, in particular the latter clause.

**Reply by hafezdivandari** [reply]
This behaviour is already documented:

https://laravel.com/docs/12.x/passport#requesting-tokens-redirecting-for-authorization

![Screenshot_20251213_221606_Chrome](https://github.com/user-attachments/assets/49d08264-6587-44b6-ab63-6820e871e716)

In addition:

![Screenshot_20251213_222031_Chrome](https://github.com/user-attachments/assets/6a6acd08-597a-483f-8340-edf3874dbe92)
